### PR TITLE
Replaced tomlc99 with tomlplusplus

### PR DESCRIPTION
**Pull request by Trial97**
This replaces the tomlc99 library with tomlplusplus as described by: https://github.com/PolyMC/PolyMC/issues/135

### CODE REVIEW

File: launcher/minecraft/mod/tasks/LocalModParseTask.cpp [hunk 1]
```diff
@@ -102,109 +93,77 @@ ModDetails ReadMCModTOML(QByteArray contents)
 {
     ModDetails details;
 
-    char errbuf[200];
-    // top-level table
-    toml_table_t* tomlData = toml_parse(contents.data(), errbuf, sizeof(errbuf));
-
-    if(!tomlData)
-    {
+    // auto tomlData = toml::parse(contents.toStdString());
```
Comment (flowln) [inline launcher/minecraft/mod/tasks/LocalModParseTask.cpp]:
```suggestion
```

**Review by Scrumplex — COMMENTED** [review]
Haven't reviewed the code yet, but wanted to point out, that [tomlplusplus is packaged](https://repology.org/project/tomlplusplus/versions) on some distros, so we should try to prefer system libs over submodules, the same way we do for QuaZip already

**Comment by Trial97** [comment]
Thanks for your comment. I updated the cmake files to use the system tomlplusplus library and fallback to bundled version otherwise.

**Comment by txtsd** [comment]
I would've preferred the formatting and actual changes be broken up into separate commits, but this should be okay for now.

File: CMakeLists.txt [hunk 2]
```diff
@@ -312,7 +312,15 @@ endif()
 add_subdirectory(libraries/rainbow) # Qt extension for colors
 add_subdirectory(libraries/LocalPeer) # fork of a library from Qt solutions
 add_subdirectory(libraries/classparser) # class parser library
-add_subdirectory(libraries/tomlc99) # toml parser
+if(NOT Launcher_FORCE_BUNDLED_LIBS)
+  find_package(tomlplusplus 3.2.0 QUIET)
+endif()
+if(NOT tomlplusplus_FOUND)
+    message(STATUS "Using bundled tomlplusplus")
+    add_subdirectory(libraries/tomlplusplus) # toml parser
```
Comment (flowln) [inline CMakeLists.txt:322]: do we need `set(BUILD_SHARED_LIBS 0)` here? when using the bundled quazip we do, so wouldn't this be the same?

**Review by flowln — COMMENTED** [review]
just a couple thingies, otherwise seems fine to me!

File: CMakeLists.txt (hunk 2 above)
Comment (Trial97) [reply CMakeLists.txt:322]: I looked this up in the cmake files and it looks like it is used by [quazip](https://github.com/stachenov/quazip/blob/master/quazip/CMakeLists.txt#L63-L76) only to `suppress dllimport`.
The [libnbtplusplus](https://github.com/PolyMC/libnbtplusplus/blob/master/CMakeLists.txt#L64-L67) uses a similar variable but is not the same.

Comment (flowln) [reply CMakeLists.txt:322]: aight, thanks for checking it :)

**Review by flowln — APPROVED** [review]
seems fine to me.

thanks!! :D

File: CMakeLists.txt [hunk 3]
```diff
@@ -312,7 +312,15 @@ endif()
 add_subdirectory(libraries/rainbow) # Qt extension for colors
 add_subdirectory(libraries/LocalPeer) # fork of a library from Qt solutions
 add_subdirectory(libraries/classparser) # class parser library
-add_subdirectory(libraries/tomlc99) # toml parser
+if(NOT Launcher_FORCE_BUNDLED_LIBS)
+  find_package(tomlplusplus 3.2.0 QUIET)
+endif()
```
Comment (Scrumplex) [inline CMakeLists.txt]: I would prefer this to be further up. Maybe right after the huge block for finding Qt (so at line 193)

**Review by Scrumplex — APPROVED** [review]
Small nitpick. Looks good overall. Thanks!

File: CMakeLists.txt (hunk 3 above)
Comment (Scrumplex) [reply CMakeLists.txt]: (Did it myself :eyes:)

**Comment by Scrumplex** [comment]
Forgot to merge this

**Pull request merged**